pandas/tests/indexes/test_setops.py

```diff
@@ -436,9 +464,20 @@ def test_intersect_unequal(self, index_flat, fname, sname, expected_name):
         # test copy.intersection(subset) - need sort for unicode and string
         first = index.copy().set_names(fname)
         second = index[1:].set_names(sname)
-        intersect = first.intersection(second).sort_values()
-        expected = index[1:].set_names(expected_name).sort_values()
-        tm.assert_index_equal(intersect, expected)
+        if (
+            len(index.values) > 0
```

Comment:
maybe a tm.check_for_this_case to de-duplicate all these?  or just a new fixture that specifically excludes the relevant case?

Reply:
thank you, I added `tm.assert_mixed_int_string_entry` and replaced the checks with it.